src/lib/share/shareItem.ts

```diff
@@ -0,0 +1,48 @@
+import { HttpStatusCode, IModify } from '@rocket.chat/apps-engine/definition/accessors';
+import { UIKitBlockInteractionContext } from '@rocket.chat/apps-engine/definition/uikit';
+
+import { TodoistApp } from '../../../TodoistApp';
+import { HttpMethod } from '../../helpers/http';
+
+interface ShareItemParams {
+  app: TodoistApp;
+  context: UIKitBlockInteractionContext;
+  modify: IModify;
+  getUrl: (id?: string) => string;
+  formatMessage: (response: any) => string;
+  method?: HttpMethod; // Optional - defaults to GET
```

Comment:
you don't have to mark it as optional in the type, the function signature already ensures that for you